### PR TITLE
implement LongArray read/write operations (fixes #1)

### LINKED ISSUE #1: Long Array type not supported
I tried to edit an NBT file with an array of longs (which is a tag type that was added in 1.12 back in 2017), but got this error:
```
Error: Unable to parse test.nbt, are you sure it's an NBT file?
        caused by: Got unknown type id c trying to read NBT compound
For help, run with --help or read the manpage.
```

When I removed the array using NBTExplorer and tried again, it loaded fine, so I know that's what's causing the problem.

### PR DESCRIPTION
**Pull request by devvmh**
NOTE: this commit doesn't work for me. With my specific unreadable NBT file, I still get a different error message:

	caused by: failed to fill whole buffer

However, it does remove the previous error I got that said

        caused by: Got unknown type id c trying to read NBT compound

### CODE REVIEW

**Comment by C4K3** [comment]
Thanks for the PR!

Seems like it should work. Would you mind adding a (failing) test case? Then I'd be happy to help debug why you're getting that error.

**Comment by devvmh** [comment]
I suspect the file is legitimately corrupted, since I've also seen errors when logging that player in.

But I have no other tools to validate with other than this one so I'm not sure what I'd do to validate the second error. I guess I should update the commit message

File: src/unstable/write.rs [hunk 1]
```diff
@@ -132,3 +133,13 @@ fn write_int_array<W: Write>(w: &mut W, val: &Vec<i32>) -> Result<()> {
 
     Ok(())
 }
+
+fn write_long_array<W: Write>(w: &mut W, val: &Vec<i64>) -> Result<()> {
+    write_long(w, val.len() as i64)?;
```
Comment (ravomavain) [inline src/unstable/write.rs:138]:
```suggestion
fn write_long_array<W: Write>(w: &mut W, val: &Vec<i64>) -> Result<()> {
    write_int(w, val.len() as i32)?;
```

File: src/unstable/read.rs [hunk 2]
```diff
@@ -191,3 +193,21 @@ fn read_int_array<R: Read>(reader: &mut R) -> Result<NBT> {
 
     Ok(NBT::IntArray(ret))
 }
+
+fn read_long_array<R: Read>(reader: &mut R) -> Result<NBT> {
+    let length = match read_long(reader)? {
+        NBT::Long(val) => val as usize,
+        _ => unreachable!(),
+    };
```
Comment (ravomavain) [inline src/unstable/read.rs:201]:
```suggestion
fn read_long_array<R: Read>(reader: &mut R) -> Result<NBT> {
    let length = match read_int(reader)? {
        NBT::Int(val) => val as usize,
        _ => unreachable!(),
    };
```

File: src/unstable/write.rs (hunk 1 above)
Comment (devvmh) [reply src/unstable/write.rs:138]: Ahh, sweet, thanks!!

**Comment by devvmh** [comment]
Fascinating, I've applied the changes above, and now I can open the file with a LongArray, but when I close it I get

```
Unable to parse edited file
	caused by: Unknown tag type LongArray
```

**Comment by devvmh** [comment]
OK, had to push one more fix (https://github.com/C4K3/nbted/commit/b4cb93d942a615b34275b1102cd0fd27d5294463) and now manual testing works.

Additionally, I've pushed an actual test (https://github.com/C4K3/nbted/commit/05fec15e14933b72324136f84b75a46d43bb04c4) asserting that LongArray works as expected. I used the actual contents of the file I was having a problem with - feel free to suggest changes to make it more generic or just change it.

**Comment by C4K3** [comment]
Thanks!

**Pull request merged**